[error tracking] Add setup instructions for ET Backend standalone (#28374)
DataDog/documentation: content/en/error_tracking/guides/enable_apm.md

---
title: Enable APM
further_reading:
- link: "/tracing"
  tag: "Documentation"
  text: "Learn about APM"
- link: "/error_tracking/guides/enable_infra"
  tag: "Guide"
  text: "Enable Infrastructure Monitoring"
---

> This guide explains how to update the Datadog Agent configuration to enable [APM][1].

Reviewer: Can you add 1-2 sentences about the benefits of enabling APM with backend error tracking?

Author: Would that answer your request?

```suggestion
[Datadog Application Performance Monitoring (APM)][1] provides deep visibility into your applications, enabling you to identify performance bottlenecks, troubleshoot issues, and optimize your services. To leverage these features on top of standalone backend error tracking, update the Datadog Agent configuration to enable APM.
```

Reviewer: I updated your suggestion slightly, but yes it's great!

   -   service_checks: false
   -   sketches: false

     # Configuration to enable the collection of errors so they show up in Error Tracking.
     apm_config:
       enabled: true
   -   error_tracking_standalone:
   -     enabled: true
   ```

3. [Restart the Agent][3].
   {{< /collapse-content >}}

{{< collapse-content title="Using Datadog tracing libraries" level="h5" >}}
For a Datadog agent configured manually for Backend Error Tracking:

1. Open the [datadog.yaml configuration file][2].
2. Remove the `core_agent` and `error_tracking_standalone` attributes:

   ```diff
   - core_agent:
   -   enabled: false
     apm_config:
   +   enabled: true
   -   error_tracking_standalone:
   -     enabled: true
   ```

3. [Restart the Agent][3].
   {{< /collapse-content >}}

[2]: /agent/configuration/agent-configuration-files
[3]: /agent/configuration/agent-commands/#restart-the-agent

{{% /tab %}}
{{% tab "Kubernetes" %}}

If your agent is deployed in Kubernetes, you need to update its configuration in Datadog Operator or Helm depending on the method you used to install the Agent.

{{< collapse-content title="Helm" level="h5" >}}
For a Datadog Agent installed with Helm:

> 1. Update your datadog-values.yaml file

Reviewer: ```suggestion
1. Update your `datadog-values.yaml` file, replacing the `site` and `env` values appropriately:
```

   ```diff
     agents:
       containers:
         agent:
           env:
             [...]
   -         - name: DD_CORE_AGENT_ENABLED
   -           value: "false"
     datadog:
   -   processAgent:
   -     enabled: false
   -     containerCollection: false
     apiKeyExistingSecret: datadog-secret
     site: <DATADOG_SITE>
     tags:
       - env:<AGENT_ENV>
     apm:
   -   errorTrackingStandalone:
   -     enabled: true
       # Required to enable Single-Step Instrumentation
       instrumentation:
         enabled: true
         libVersions:
           java: "1"
           dotnet: "3"
           python: "2"
           js: "5"
           php: "1"
   ```

> 2. After making your changes, upgrade your Datadog Helm chart

Reviewer: ```suggestion
2. After making your changes, upgrade your Datadog Helm chart:
```

   ```shell
   helm upgrade -f datadog-values.yaml datadog-agent datadog/datadog
   ```
{{< /collapse-content >}}

> {{< collapse-content title="Datadog Operator" level="h5" >}}

Reviewer: We can add this tab when we've added support.

Author: ```suggestion

```

> Coming soon

Reviewer: ```suggestion
```

> {{< /collapse-content >}}

Reviewer: ```suggestion
```
